### DISCUSSION TITLE
Streaming plasma in 1D simulation

### DISCUSSION THREAD

**Discussion by Misery7100** [epochpic/epoch#442] [Q&A] [answered]
Hello EPOCH community,

I'm trying to reproduce 1D calculations of streaming overdense plasma interaction with femtosecond impulse. Previously it was done using LPIC++ code (e.g. https://github.com/ruddyscent/lpic) and there is a feature to simulate it using specified angle of light source out of 1D modeling box (check attached picture, it's 50th page of lpic++ manual, reachable by link previously provided).

![lpic_fig29](https://user-images.githubusercontent.com/68801526/217100302-de4121bc-6c51-48b4-b3c0-55101676fb49.png)

I tried to to something using 2D code and injector from y plane, but computation time is much higher even with 3 cells by y only, so I don't sure it was properly configured. I will be very appreciated for help with the issue

Best,
Lev

**Comment by Status-Mirror** [comment, marked as answer]
Hi Lev,

I would expect a 2D-PIC code to have a much slower run-time compared to an equivalent 1D simulation. Interpolation of fields to the particle position becomes more complicated, so it wouldn't just be (1D-runtime)*(number of cells in y).

For help in setting up a 2D input deck with an angled incident laser, consider starting with the example input deck given at the bottom of this page: https://epochpic.github.io/documentation/examples/basic_examples.html

Hope this helps,
Stuart

**Comment by Misery7100** [comment]
Hello Stuart,

Thanks very much for your answer. Of course, I understand 2D complexity in comparison with 1D, just it was not kinda 3 times more, but something like squared time, so I found it a bit strange for quite simple task. Anyway, seems like it's impossible to emulate streaming plasma using 1D only, right? Because I checked all documentation, but can't found parameters to emulate y velocity for particles in 1D and injection at the same time

Best,
Lev

**Reply by Status-Mirror** [reply]
Hey Lev,

If you want to set up a 1D simulation like Fig 29. (M) in your attached figure, EPOCH can do this.

You're right, EPOCH unfortunately doesn't have a simple "shift frame" key for the input deck, but we can still set up this simulation. For the laser, you can adjust the laser properties in the laser block (adding the $\cos(\alpha)$ term to frequency and wavenumber). For the streaming plasma, you can use the `drift_y` key in the [species block](https://epochpic.github.io/documentation/input_deck/input_deck_species.html) to add a streaming momentum to each particle. This should provide the functionality you seek.

There are still advantages to modelling this in 2D, as this would allow you to model a tight laser focus and electron divergence inside the plasma - but for quick test simulations, 1D should work fine.

Hope this helps,
Stuart

**Reply by Misery7100** [reply]
Stuart, you made my day. Thanks so much!
